sema4ai/vscode-client/src/robo/actionPackage.ts

```diff
@@ -384,3 +409,250 @@ async function getTemplate(actionServerLocation: string) {
     const selectedTemplate = templates.find((template) => template.description === selectedItem);
     return selectedTemplate?.name;
 }
+
+const chooseOrganization = async (
+    organizations: ActionServerListOrganizationsOutput
+): Promise<ActionServerOrganization> => {
+    const organization = await window.showQuickPick(
+        organizations.map((organization) => organization.name),
+        {
+            "canPickMany": false,
+            "placeHolder": "Select the organization to publish package to",
+            "ignoreFocusOut": true,
+        }
+    );
+
+    return organizations.find((org) => org.name == organization);
+};
+
+const buildPackage = async (actionServerLocation: string, workspaceDir: string, outputDir: string): Promise<string> => {
+    const result: ActionResult<ActionServerPackageBuildOutput> = await commands.executeCommand(
+        roboCommands.SEMA4AI_ACTION_SERVER_PACKAGE_BUILD_INTERNAL,
+        {
+            action_server_location: actionServerLocation,
+            workspace_dir: workspaceDir,
+            output_dir: outputDir,
+        }
+    );
+    if (!result.success) {
+        window.showErrorMessage(`Failed to upload action package: ${result.message}`);
+        return;
+    }
+
+    return result.result.package_path;
+};
+
+const askUserForChangelogInput = async (): Promise<string> => {
+    return window.showInputBox({
```

Comment:
I guess this isn't ideal (maybe I'm wrong, but from what I remember `showInputBox` is only a single-line entry, no?)

I think the initial integration can happen as is, but I think the proper approach here would be having a custom webview for this -- as is done for the submit issue (which went the same way of starting with a series of pop-ups and evolved to a webview).

-- so, the upload would actually open the webview and then the user could do all of the selections needed (and add the changelog) on the webview instead of a series of pop-ups...

Reply:
I agree that a webview would be better, but that can come later in the game.
Right now this implementation is enough.

We should create a Linear Task for this & add it to the backlog.

Reply:
Sure, though the ACE side UI doesn't really support multiline changelog entries either.